plugins/search-backend/src/service/AuthorizedSearchEngine.ts

```diff
@@ -197,7 +198,11 @@ export class AuthorizedSearchEngine implements SearchEngine {
           const permission = this.types[result.type]?.visibilityPermission;
           const resourceRef = result.document.authorization?.resourceRef;
 
-          if (!permission || !resourceRef) {
+          if (
+            !permission ||
+            !isResourcePermission(permission) ||
+            !resourceRef
```

Comment:
Would this result from an incorrectly indexed document? Should we log a warning in that case?

Reply:
We only reach this point in the code for types where the initial authorization returned `CONDITIONAL` -- `ALLOW`s return early immediately above, and types where the decision was `DENY` [get filtered out entirely when querying](https://github.com/backstage/backstage/pull/10068/files#diff-0c22d1c63b390276b4bf3f1c3fdd38c86fb3c0cfbfaeb66b51a6afe1825799b1R118).

This means the call to `isResourcePermission` here is mostly about narrowing the types, since the only way for it to be false is if the `PermissionPolicy` returned a `CONDITIONAL` decision for a non-resource permission, which can't happen - it would throw an error [here](https://github.com/backstage/backstage/blob/master/plugins/permission-backend/src/service/router.ts#L112).

Maybe we could throw here too for good measure though - something like `throw new Error('Unexpected CONDITIONAL decision for non-resource permission - this should never happen!')`. WDYT?

Reply:
Gotcha. Throwing seems fair in this case since the permission backend should never return conditions when a resource ref is supplied

Reply:
Yeah, I think it's clearer. Done in 7471d5e55e.